src/NamingStrategy/MapNamingStrategy.php

```diff
@@ -9,83 +9,100 @@
 
 namespace Zend\Hydrator\NamingStrategy;
 
-use Zend\Hydrator\Exception\InvalidArgumentException;
+use Zend\Hydrator\Exception;
 
 use function array_flip;
 use function array_key_exists;
 use function array_walk;
 use function is_int;
 use function is_string;
+use function sprintf;
 
-class MapNamingStrategy implements NamingStrategyInterface
+final class MapNamingStrategy implements NamingStrategyInterface
 {
     /**
-     * Map for hydrate name conversion.
-     *
      * @var string[]
      */
-    protected $mapping = [];
+    private $extractionMap = [];
 
     /**
-     * Reversed map for extract name conversion.
-     *
      * @var string[]
      */
-    protected $reverse = [];
+    private $hydrationMap = [];
 
     /**
-     * @param string[]      $mapping Map for name conversion on hydration
-     * @param null|string[] $reverse Reverse map for name conversion on extraction
+     * @param null|string[] $hydrationMap A map of string keys and values for
```

Comment:
In this case, the `null|array<string, string>` notation is quite important, as the type of key is relevant to us.